pkg/controller/resource/resource.go

```diff
@@ -34,20 +34,31 @@ func CheckChaosAnnotation(ce *chaosTypes.EngineInfo) (*chaosTypes.EngineInfo, er
 		if err != nil {
 			return ce, fmt.Errorf("resource type 'statefulset', err: %+v", err)
 		}
+	case "daemonset", "daemonsets":
+		ce, err = CheckDaemonSetAnnotation(clientSet, ce)
+		if err != nil {
+			return ce, fmt.Errorf("resource type 'daemonset', err: %+v", err)
+		}
 	default:
 		return ce, fmt.Errorf("resource type '%s' not supported for induce chaos", ce.AppInfo.Kind)
 	}
 	return ce, nil
 }
 
-// ValidateAnnotation will verify the validation require for induce chaos
-func ValidateAnnotation(annotationValue string, chaosCandidates int) (int, error) {
+// CountTotalChaosEnabled will count the number of chaos enabled applications
+func CountTotalChaosEnabled(annotationValue string, chaosCandidates int) int {
 	if annotationValue == ChaosAnnotationValue {
 		chaosCandidates++
-	} else if chaosCandidates > 1 {
-		return chaosCandidates, errors.New("too many chaos candidates with same label, either provide unique labels or annotate only desired app for chaos")
-	} else if chaosCandidates == 0 {
-		return chaosCandidates, errors.New("no chaos-candidate found")
 	}
-	return chaosCandidates, nil
+	return chaosCandidates
```

Comment:
This will always keep on increasing chaosCandidates without checking if they are greater than 1 and check this condition at the end. 
However, ideally we should give an error as soon as they are detected greater than 1 leaving all the remaining resources.

Reply:
Yes!!
We can put check without increasing also, Will do in the next PR, Just keeping this PR related to `daemonset` only

Reply:
Okay..great

Reply:
Lets create an issue to track this.